tap-agent/src/agent/sender_account.rs

```diff
@@ -209,6 +256,83 @@ impl Actor for SenderAccount {
                     }
                 });
 
+        let myself_clone = myself.clone();
+        let pgpool_clone = pgpool.clone();
+        let _escrow_account_monitor = escrow_accounts.clone().pipe_async(move |escrow_account| {
+            let myself = myself_clone.clone();
+            let pgpool = pgpool_clone.clone();
+            // get balance or default value for sender
+            // this balance already takes into account thawing
+            let balance = escrow_account
+                .get_balance_for_sender(&sender_id)
+                .unwrap_or_default();
+
+            #[derive(Deserialize)]
+            struct Transaction {
+                #[serde(rename="allocationID")]
+                allocation_id: String,
+            }
+
+            #[derive(Deserialize)]
+            struct Response {
+                transactions: Vec<Transaction>
+            }
+
+            async move {
+                let last_non_final_ravs =
+                    sqlx::query!(
+                        r#"
+                            SELECT allocation_id, value_aggregate FROM scalar_tap_ravs WHERE sender_address = $1 AND last AND NOT final;
```

Comment:
Super long strings like this break cargo-fmt completely on the whole file.
Gotta make sure string / comments don't push the line over 100 characters.
That's why the file is actually not formatted...
```suggestion
                            SELECT allocation_id, value_aggregate 
                            FROM scalar_tap_ravs 
                            WHERE sender_address = $1 AND last AND NOT final;
```